Documentation/Evolution/CharacterClasses.md

```diff
@@ -0,0 +1,415 @@
+
+</details>
+
+### Control characters: `\t`, `\r`, `\n`, `\f`, `\0`, `\e`, `\a`, `\b`, `\cX`
+
+We propose the following names and meanings for these escaped literals representing specific control characters:
+
+```swift
+extension Character {
+  /// A horizontal tab character, `CHARACTER TABULATION` (U+0009).
+  public static var tab: Character { get }
+
+  /// A carriage return character, `CARRIAGE RETURN (CR)` (U+000D).
+  public static var carriageReturn: Character { get }
+
+  /// A line feed character, `LINE FEED (LF)` (U+000A).
+  public static var lineFeed: Character { get }
+
+  /// A form feed character, `FORM FEED (FF)` (U+000C).   
+  public static var formFeed: Character { get }
+
+  /// A NULL character, `NUL` (U+0000).   
+  public static var nul: Character { get }
+
+  /// An escape control character, `ESC` (U+001B).
+  public static var escape: Character { get }
+
+  /// A bell character, `BEL` (U+0007).
+  public static var bell: Character { get }
+
+  /// A backspace character, `BS` (U+0008).
+  public static var backspace: Character { get }
+
+  /// Returns a control character with the given value, Control-`x`.
+  ///
+  /// This method returns a value only when you pass a letter in 
+  /// the ASCII range as `x`:
+  ///
+  ///     if let ch = Character.control("G") {
+  ///         print("'ch' is a bell character", ch == Character.bell)
+  ///     } else {
+  ///         print("'ch' is not a control character")
+  ///     }
+  ///     // Prints "'ch' is a bell character: true"
+  ///
+  /// - Parameter x: An upper- or lowercase letter to derive
+  ///   the control character from.
+  /// - Returns: Control-`x` if `x` is in the pattern `[a-zA-Z]`;
+  ///   otherwise, `nil`.
+  public static func control(_ x: Unicode.Scalar) -> Character?
+}
+
+extension Unicode.Scalar {
+  /// Same as above...
+}
+```
+
+**TODO**: What about `\r\n` in grapheme semantic mode?
+
+_<details><summary>Rationale</summary>_
+
+This approach simplifies the use of some common control characters, while making the rest available through a method call.
+
+</details>
+
+
+
+### Unicode named values and properties: `\N`, `\p`, `\P`
+
+`\N{NAME}` matches a Unicode scalar value with the specified name. `\p{PROPERTY}` and `\p{PROPERTY=VALUE}` match a Unicode scalar value with the given Unicode property (and value, if given). 
+
+While most Unicode-defined properties can only match at the Unicode scalar level, some are defined to match an extended grapheme cluster. For example, `/\p{RGI_Emoji_Flag_Sequence}/` will match any flag emoji character, which are composed of two Unicode scalar values.
+
+`\P{...}` matches the inverse of `\p{...}`.
+
+Most of this functionality is already provided inside `Unicode.Scalar.Properties`, and we propose to round out Swift's current support with API to access the Unicode script(s) for a `Unicode.Scalar` or `Character`. (API TBD)
+
+**TODO**: Check with Alejandro that the code size impact is reasonable
```

Comment:
**TODO** :-)

Reply:
I think we can just delete this TODO. It seems like we need to provide this functionality so we'll just need to strike the right code-size/runtime efficiency balance in implementing it.